### PR TITLE
Dev feefw

### PR DESCRIPTION
**Pull request by a-piasta**

### CODE REVIEW

**Comment by Lorak-mmk** [comment]
Please change commit messages...

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ SET(CMAKE_CXX_STANDARD 17)
 
 project(EchoConnect VERSION 0)
 
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -fno-strict-aliasing")
```
Comment (Lorak-mmk) [inline CMakeLists.txt:9]: Why is that required?

File: libecho/src/Echo.cc [hunk 2]
```diff
@@ -8,10 +8,11 @@
 #include <QCoreApplication>
 #include <QDebug>
 
-static constexpr int windowSize = 200;
-static constexpr int loFreq = 19000;
-static constexpr int hiFreq = 20000;
-static constexpr double magLimit = 300.0 / windowSize;
+static constexpr int windowSize = 300;
+static constexpr int loFreq = 14500;
+static constexpr int hiFreq = 15000;
```
Comment (Lorak-mmk) [inline libecho/src/Echo.cc:13]: Is it necessary to use those frequencies? One of project's basic ideas is to use high frequencies that human can barely hear.

File: libecho/tests/tmp_test.cpp [hunk 3]
```diff
@@ -3,26 +3,72 @@
...
+    for (int i = 0; i < 256; i++) {
         vbuffer.push_back(i);
     }
     Echo echo;
     echo.send(vbuffer);
-}
-
-TEST(test_receive, tmp_test) {
-    Echo echo;
+}*/
```
Comment (Lorak-mmk) [inline libecho/tests/tmp_test.cpp:36]: If code is commented out, maybe consider deleting it?

File: libecho/tests/tmp_test.cpp [hunk 4]
```diff
@@ -3,26 +3,72 @@
...
+        echo.send(vbuff);
+        
+        int wstatus;
+        pid_t p = wait(&wstatus);
+        assert(p == pid);
+        if (WEXITSTATUS(wstatus) > 90) exit(0);
+        else exit(1);
     }
-    printf("\n");
 }
```
Comment (Lorak-mmk) [inline libecho/tests/tmp_test.cpp:74]: Read this https://github.com/google/googletest/blob/master/googletest/docs/primer.md and rewrite this code

**Review by Lorak-mmk — APPROVED** [review]
Please fix test code and add more tests. Also change commit messages.
Approved only because there is no time...

File: CMakeLists.txt (hunk 1 above)
Comment (a-piasta) [reply CMakeLists.txt:9]: reinterpret_cast w echo.cc

File: libecho/src/Echo.cc (hunk 2 above)
Comment (a-piasta) [reply libecho/src/Echo.cc:13]: testowałem na moim kompie różne; 19k i 20k, jakie były poprzednio, tworzyły błędy, zaś to dawało dobre wyniki

**Review by Riyria-was-taken — APPROVED** [review]
:v
